android/src/main/java/com/reactnativeimageresizer/ImageResizer.java

```diff
@@ -316,39 +315,57 @@ public static boolean copyExif(Context context, Uri imageUri, String dstPath){
   /**
    * Get orientation by reading Image metadata
    */
-  public static int getOrientation(Context context, Uri uri) {
+  public static Matrix getOrientationMatrix(Context context, Uri uri) {
     try {
       // ExifInterface(InputStream) only exists since Android N (r24)
       if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
         InputStream input = context.getContentResolver().openInputStream(uri);
         ExifInterface ei = new ExifInterface(input);
-        return getOrientation(ei);
+        return getOrientationMatrix(ei);
       }
       File file = getFileFromUri(context, uri);
       if (file.exists()) {
         ExifInterface ei = new ExifInterface(file.getAbsolutePath());
-        return getOrientation(ei);
+        return getOrientationMatrix(ei);
       }
     } catch (Exception ignored) { }
 
-    return 0;
+    return new Matrix();
   }
 
   /**
    * Convert metadata to degrees
    */
-  public static int getOrientation(ExifInterface exif) {
-    int orientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL);
+  public static Matrix getOrientationMatrix(ExifInterface exif) {
+    Matrix matrix = new Matrix();
+    int orientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_UNDEFINED);
     switch (orientation) {
+      case ExifInterface.ORIENTATION_FLIP_HORIZONTAL:
+        matrix.setScale(-1, 1);
+        break;
+      case ExifInterface.ORIENTATION_TRANSPOSE:
+        matrix.setRotate(90);
+        matrix.postScale(-1, 1);
+        break;
       case ExifInterface.ORIENTATION_ROTATE_90:
-        return 90;
+        matrix.setRotate(90);
+        break;
+      case ExifInterface.ORIENTATION_FLIP_VERTICAL:
+        matrix.setRotate(180);
+        matrix.postScale(-1, 1);
```

Comment:
I think you can simplify with:
```suggestion
        matrix.setScale(1,-1)
```

Tell me if i'm wrong

Reply:
I haven't test that but I don't think it will work, because this will flip the image on `y axis`, So Imagine the `F` rotated 180, then imagine `F` flipped on `y` axis, it will look at the other way ( flipped 180 degree ).

Maybe if its `matrix.setScale(-1,-1)` this should work.

But I will not prefer this as it will not have any different neither optimize anything, And the current code will be much clear for any future developer.